### PR TITLE
Add JsonSchemaEntities for sttp client

### PR DESCRIPTION
**Pull request by krzemin**

### CODE REVIEW

**Comment by Krever** [comment]
Do you think it would be possible not to rely on particular json implementation (`circe`) but allow user to use sttp client with whatever json lib they already have?

File: sttp/client/src/main/scala/endpoints/sttp/client/JsonSchemaEntities.scala [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+  def jsonRequest[A : JsonRequest](docs: Documentation = None): RequestEntity[A] = { (a, request) =>
+    request
+      .body(JsonSchema.toCirceEncoder[A].apply(a).noSpaces)
+      .contentType("application/json")
+  }
+
+  def jsonResponse[A : JsonResponse](docs: Documentation = None): Response[A] = new SttpResponse[A] {
+    override type ReceivedBody = Either[Exception, A]
+    override def responseAs = sttp.asString.map { str =>
```
Comment (julienrf) [inline sttp/client/src/main/scala/endpoints/sttp/client/JsonSchemaEntities.scala]: What happens if the response entity is binary?

File: sttp/build.sbt [hunk 2]
```diff
@@ -14,4 +15,8 @@ val `sttp-client` =
         "com.typesafe.akka" %% "akka-stream" % "2.5.18" % Test
       )
     )
-    .dependsOn(`algebra-jvm` % "compile->compile;test->test", `algebra-playjson-jvm` % "test->test")
+    .dependsOn(
+      `algebra-jvm` % "compile->compile;test->test",
+      `algebra-playjson-jvm` % "test->test",
+      `json-schema-circe-jvm`
+    )
```
Comment (julienrf) [inline sttp/build.sbt]: Should we introduce a `sttp-client-circe` module instead of adding a dependency to the `json-schema-circe` module here?

**Comment by julienrf** [comment]
Thanks @krzemin for your initial work. I’m closing the PR due to inactivity. I’m happy to review and merge an updated version of the contribution.

**Pull request closed without merging**